main.rkt

```diff
@@ -1,18 +1,28 @@
 #lang racket
 
 (require "enode.rkt" "egraph.rkt" "ematch.rkt" "extraction.rkt")
+(require debug/repl)
 
-(provide make-regraph regraph-cost regraph-count regraph-extract
-         rule-phase precompute-phase prune-phase extractor-phase)
 
-(struct regraph (egraph extractor ens limit))
+(module+ test (require rackunit math/base))
 
-(define (make-regraph exprs #:limit [limit #f])
+(provide make-regraph regraph-cost regraph-count regraph-extract regraph-limit
+         rule-phase precompute-phase prune-phase extractor-phase find-matches-time
+         regraph-match-count regraph-eclass-count rebuild-phase)
+
+(struct regraph (egraph extractor ens limit match-count rebuilding-enabled?) #:mutable)
```

Comment:
Added a flag for rebuilding, which is enabled by default. Also added match-count that counts the total number of matches ever applied.